packages/edit-navigation/src/store/test/actions.js

```diff
@@ -151,289 +57,53 @@ describe( 'createMissingMenuItems', () => {
 			],
 		};
 
-		const mapping = {
-			87: 'navigation-block-client-id',
-			100: 'navigation-link-block-client-id-1',
-		};
-
 		const menuItemPlaceholder = {
-			id: 101,
+			id: 87,
 			title: {
 				raw: 'Placeholder',
 				rendered: 'Placeholder',
 			},
 		};
 
-		const menuItems = [
-			{
-				id: 100,
-				title: {
-					raw: 'wp.com',
-					rendered: 'wp.com',
-				},
-				url: 'http://wp.com',
-				menu_order: 1,
-				menus: [ 1 ],
-			},
-			{
-				id: 101,
-				title: {
-					raw: 'wp.org',
-					rendered: 'wp.org',
-				},
-				url: 'http://wp.org',
-				menu_order: 2,
-				menus: [ 1 ],
-			},
-		];
-
-		const action = createMissingMenuItems( post );
-
-		expect( action.next( post ).value ).toEqual(
-			getMenuItemToClientIdMapping( post.id )
-		);
-
-		expect( action.next( mapping ).value ).toEqual(
-			apiFetch( {
-				path: `/__experimental/menu-items`,
-				method: 'POST',
-				data: {
-					title: 'Placeholder',
-					url: 'Placeholder',
-					menu_order: 0,
-				},
-			} )
-		);
-
-		expect( action.next( menuItemPlaceholder ).value ).toEqual(
-			resolveMenuItems( post.meta.menuId )
-		);
+		const dispatch = jest
+			.fn()
+			.mockReturnValueOnce( {} )
+			.mockReturnValueOnce( menuItemPlaceholder )
+			.mockReturnValueOnce( { ...menuItemPlaceholder, id: 88 } );
+		const registryDispatch = {
+			__unstableAcquireStoreLock: jest.fn(),
+			__unstableReleaseStoreLock: jest.fn(),
+		};
+		const registry = {
+			dispatch: jest.fn( () => registryDispatch ),
+		};
 
-		expect( action.next( menuItems ).value ).toEqual(
-			dispatch(
-				'core',
-				'receiveEntityRecords',
-				'root',
-				'menuItem',
-				[ ...menuItems, menuItemPlaceholder ],
-				menuItemsQuery( post.meta.menuId ),
-				false
-			)
-		);
+		await createMissingMenuItems( post )( { registry, dispatch } );
 
-		expect( action.next().value ).toEqual( {
+		expect( dispatch ).toHaveBeenCalledWith( {
 			type: 'SET_MENU_ITEM_TO_CLIENT_ID_MAPPING',
 			postId: post.id,
 			mapping: {
 				87: 'navigation-block-client-id',
-				100: 'navigation-link-block-client-id-1',
-				101: 'navigation-link-block-client-id-2',
+				88: 'navigation-block-client-id2',
 			},
 		} );
-
-		expect( action.next( [] ).done ).toBe( true );
 	} );
 } );
 
-describe( 'saveNavigationPost', () => {
```

Comment:
I removed these `saveNavigationPost` tests for now as they are customizer-specific. Let's discuss the ideas in this PR and then double down with more relevant tests.

Reply:
We could use unit tests for the async REST API-based version, but I think it's okay to follow-up with that to avoid inflating this PR.